### PR TITLE
'Solution'

### PR DESCRIPTION
**Pull request by LaskoA**
Hello,

I guess there are 2 issues with my solution and I need advices to fix them:

1) parse.py file includes requested functions but I have doubts that "type" output is OK. Could you please tell me how to check type properly?

2) optional_task.py gets info about modules, topics and duration - should it be a seperate function? Or should I combine it with parse_single_course function somehow? Did I choose the right way to get needed data?

Thank you :)

### CODE REVIEW

File: app/parse.py [hunk 1]
```diff
@@ -14,5 +19,29 @@ class Course:
...
 
 
+def parse_single_course(course_soup) -> Course:
+    single_course = dict(
+        name=course_soup.select_one(".typography_landingH3__vTjok").text,
+        short_description=course_soup.select_one
+        (".CourseCard_flexContainer__dJk4p > p"
+         ).text,
+        type=CourseType("part-time") if "Вечірній" in str(
+            course_soup.select_one(".typography_landingH3__vTjok").text)
```
Comment (Abnormaltype) [inline app/parse.py:29]: this is OK, but it would be better to create a variable, to store this value

File: app/optional_task.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    num_modules = soup.select_one(
+        ".CourseModulesHeading_modulesNumber__GNdFP > p"
+    ).text
+    num_topics = soup.select_one(
+        ".CourseModulesHeading_topicsNumber__PXMnR > p"
+    ).text
+    duration = soup.select_one(
+        ".CourseModulesHeading_courseDuration__f_c3H > p"
+    ).text
+
```
Comment (Abnormaltype) [inline app/optional_task.py:29]: it's okay

File: app/optional_task.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
+from bs4 import BeautifulSoup
+import requests
+
+root = "https://mate.academy/"
```
Comment (Abnormaltype) [inline app/optional_task.py:4]: should be constant

File: app/optional_task.py [hunk 4]
```diff
@@ -0,0 +1,30 @@
+from bs4 import BeautifulSoup
+import requests
+
+root = "https://mate.academy/"
+website = f'{root}/courses'
```
Comment (Abnormaltype) [inline app/optional_task.py:5]: should be constant

File: app/optional_task.py [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+root = "https://mate.academy/"
+website = f'{root}/courses'
+result = requests.get(website)
+content = result.text
+soup = BeautifulSoup(content, "html.parser")
+
+box = soup.find(class_='cell large-6 large-offset-1 mb-32')
+
+links = [link['href'] for link in box.find_all('a', class_='mb-16', href=True)]
```
Comment (Abnormaltype) [inline app/optional_task.py:12]: should be constant

File: app/optional_task.py (hunk 2 above)
Comment (Abnormaltype) [inline app/optional_task.py:29]: should be function

**Review by Abnormaltype — CHANGES_REQUESTED** [review]

**Comment by LaskoA** [comment]
@Abnormaltype, please check now after fixes. Regarding constants in optional_task.py: it`s related only to ROOT and WEBSITE, right? Thank you :)

File: app/optional_task.py [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+import requests
+
+ROOT = "https://mate.academy/"
+WEBSITE = f'{ROOT}/courses'
+
+result = requests.get(WEBSITE)
+content = result.text
+soup = BeautifulSoup(content, "html.parser")
+box = soup.find(class_='cell large-6 large-offset-1 mb-32')
+links = [link['href'] for link in box.find_all('a', class_='mb-16', href=True)]
```
Comment (Abnormaltype) [inline app/optional_task.py:11]: yes, it should be in the other function or should be in the class.
Ex.
```
def func() -> list:
    some code
    return links
```

**Review by Abnormaltype — CHANGES_REQUESTED** [review]

**Comment by LaskoA** [comment]
I pyt everything inside course_detailed_info - is that OK? Thank you.

**Review by Abnormaltype — APPROVED** [review]
yes, okay) approved, but use `""` instead of `''`

**Pull request closed without merging**